clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -399,9 +399,11 @@ private Optional<MetadataResponse.PartitionMetadata> updateLatestMetadata(
                 // Between the time that a topic is deleted and re-created, the client may lose track of the
                 // corresponding topicId (i.e. `oldTopicId` will be null). In this case, when we discover the new
                 // topicId, we allow the corresponding leader epoch to override the last seen value.
-                log.info("Resetting the last seen epoch of partition {} to {} since the associated topicId changed from {} to {}",
-                         tp, newEpoch, oldTopicId, topicId);
-                lastSeenLeaderEpochs.put(tp, newEpoch);
+                if (!Objects.equals(currentEpoch, newEpoch)) {
```

Comment:
I'm wondering though if using a primitive and an Integer is ok here. Maybe we can just compare the value of currentEpoch?